javascripts/discourse/components/search-banner.js

```diff
@@ -31,9 +31,9 @@ export default Component.extend({
     const showFor = settings.show_for;
     if (showFor == "everyone") {
       return true;
-    } else if (showFor == "logged_out" && !currentUser) {
+    } else if (showFor == "logged_out" && !this.currentUser) {
       return true;
-    } else if (showFor == "logged_in" && currentUser) {
+    } else if (showFor == "logged_in" && this.currentUser) {
       return true;
```

Comment:
I think the best solution here would be to pass the `currentUser` object to the function. `discourseComputed` passes it automatically, so we just need to add it as an argument to the function like:

```diff
  @discourseComputed("currentUser")
- displayForUser() {
+ displayForUser(currentUser) {
    const showFor = settings.show_for;
    if (showFor == "everyone") {
      return true;
    } else if (showFor == "logged_out" && !this.currentUser) {
      return true;
    } else if (showFor == "logged_in" && this.currentUser) {
      return true;
    }
    return false;
  }
```
This is (very slightly) more efficient than calling `this.currentUser` multiple times.